### DISCUSSION TITLE
How to upload images as 3D posters to be showcased on the wall?

### DISCUSSION THREAD

**Discussion by mihirm3hub** [google/model-viewer#2697] [Q&A]
I was checking out echo AR & you can apparently upload images to it & they seem to be using model viewer as well so I was curious to understand how can I replicate something similar to showcase my art/digital portraits.
Here is an example from echo AR :  [Wall Poster](https://go.echoar.xyz/v3yq )
where an image is uploaded to the model viewer and can be seen in 3d or AR. The wall placement is documented but I am unable to understand the upload the image as 3d part.

**Comment by ThorstenBux** [comment]
Hi @mihirm3hub, is this what you mean: https://webar.tripod-digital.co.nz/?exp=https://storage.googleapis.com/ar-tee.appspot.com/art/brycegallery/bryce_gallery

It is also powered by an image of the artwork.
However, this is something outside of modelViewer and needs to be solved in your part of the code. Ping me if you like more details :)

**Reply by mihirm3hub** [reply]
hi @ThorstenBux Thanks for your example, It is similar to what I am looking for. But I also want the ability to have an option to upload images, may be as textures. check this out https://www.cameraah.com/ If you can scroll down to the Cameraah in action tab and choose the mobile case there you can find an option to upload image as texture on the mobile case, I want to recreate something similar and instead of mobile case I want to use simple plane so it looks like a portrait on the wall.

**Reply by ThorstenBux** [reply]
Yes, there are always two parts to it. One is the viewer, which I shared, and one is the editor which I have as well. You would need to implement both and have some code glues it together for your case. (Mapps the jpeg to the texture of your wall art, creates a wall tracking object, etc.) Or you use platforms that already have that. But as said that is all application logic and not related to modelviewer itself. My solution is a subscription service but I gather that you want to implement it yourself?

**Reply by mihirm3hub** [reply]
@ThorstenBux  yes sir, I am a student and I may not be able to afford any subscription service as of now so I am looking for something I can recreate myself. Any suggestions on that front? any example or code snippet that I can make use of?

**Comment by ThorstenBux** [comment]
You can start by learning how to dynamically add textures to 3D models. Don’t have an example handy, though.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by mihirm3hub** [reply]
I am already looking into it but not much success :(

**Reply by mihirm3hub** [reply]
@ThorstenBux You did say that you already have  a subscription service which can handle this can I have a peek at it and the code if it is possible and recreate it on a smaller scale?

**Reply by elalish** [reply]
Have you  looked at this example for adding textures? https://modelviewer.dev/examples/scenegraph/#createTexturesExample

If you make a 3D model with a simple quad with UVs, you should be able to apply your texture images to it dynamically similarly to this example. You can export that simple quad by adding a  plane here: https://threejs.org/editor/

**Reply by mihirm3hub** [reply]
Hello @elalish thanks for the suggestions, I will check it out. I saw some posts where it was suggested to convert images to base64URL to load it dynamically is that some way to go?

**Reply by mihirm3hub** [reply]
@elalish I tried doing what you have suggested, but I am getting this error: Not allowed to load local resource: file:///C:/fakepath/amsterdam.jpg
here is a simple glitch to checkout :  https://glitch.com/edit/#!/upload-learn

**Reply by elalish** [reply]
This is the web! You're not allowed to access your file system directly. On Glitch you need to upload any files to your assets directory and get their URLs from there.
